repos/spack_repo/builtin/packages/gromacs/package.py

```diff
@@ -226,9 +226,19 @@ class Gromacs(CMakePackage, CudaPackage):
     )
 
     variant("openmp", default=True, description="Enables OpenMP at configure time")
+
+    # When using apple-clang version 15.x or newer, need to use the llvm-openmp library
+    # We also protect with version 2025+ as there seems to be a CMake bug with
+    # Apple Clang and OpenMP that is fixed in 2025
+    depends_on("llvm-openmp", when="@2025: %apple-clang@15: +openmp", type=("build", "run"))
```

Comment:
Syntax changed so you want the variant to be associated with this package and not the `apple-clang` compiler:

```suggestion
    depends_on("llvm-openmp", when="@2025: +openmp %apple-clang@15:", type=("build", "run"))
```

Reply:
`spack spec gromacs` succeeds with this change.